### PR TITLE
Add workflow for scheduling openQA test inside action

### PR DESCRIPTION
**Pull request by baierjan**
Reference: https://progress.opensuse.org/issues/126950

### CODE REVIEW

**Review by Martchus — COMMENTED** [review]
Strange, this didn't generate any jobs: https://openqa.opensuse.org/admin/productlog?id=312761

I'm not sure why; I couldn't spot an obvious mistake.

I'm wondering whether `openqa-cli schedule …` should have an option to require a minimum number of jobs (so not executing any jobs wouldn't lead to passing CI checks).

**Comment by baierjan** [comment]
> Strange, this didn't generate any jobs: https://openqa.opensuse.org/admin/productlog?id=312761

Yes, I am wondering as well. If I omit the group override, it works correctly.

> option to require a minimum number of jobs

That would be indeed useful.

File: .github/workflows/openqa.yml
Comment (okurz) [reply .github/workflows/openqa.yml:31]: I see.

@Martchus this line shouldn't be needed, is it?

Comment (Martchus) [reply .github/workflows/openqa.yml:31]: This is *not* the default. The default is `/api/v1` (which is an absolute path and not a relative one).

This line is only needed if `OPENQA_HOST` points to an openQA instance that is made available under a sub path, e.g. in my initial tests I have been using `OPENQA_HOST=https://martchus.no-ip.biz/openqa/sometoken` to have a test instance that is publicly available and has the changes I needed. I suppose we can remove the parameter although it shouldn't hurt in any other case. (If we are sure it doesn't cause problems in any other case we could actually make it the default.)

Comment (okurz) [reply .github/workflows/openqa.yml:31]: I would like to keep the example in os-autoinst-distri-example as simple as possible. Can you delete it there and maybe see more that can be removed or simplified?

Comment (Martchus) [reply .github/workflows/openqa.yml:31]: PR: https://github.com/os-autoinst/os-autoinst-distri-example/pull/19

I don't think it makes sense to remove any of the other parameters. Of course we could avoid e.g. `_GROUP=0` but it is actually nice to show best practices - even though if it means the example is not as simple as possible.

Comment (okurz) [reply .github/workflows/openqa.yml:31]: Fine, was just asking, maybe you had more ideas

Comment (kalikiana) [reply .github/workflows/openqa.yml:31]: @baierjan Are you removing the `--apibase` switch then?

Comment (baierjan) [reply .github/workflows/openqa.yml:31]: @kalikiana Afaik it is not the default (yet).

Comment (baierjan) [reply .github/workflows/openqa.yml]: Ok. Done.

**Review by okurz — CHANGES_REQUESTED** [review]
If I understand this correctly this will always fail for every pull request that is created from a fork, right? Then we should not merge this.

**Comment by okurz** [comment]
Then consider https://progress.opensuse.org/issues/126950 blocked by https://progress.opensuse.org/issues/128360

**Comment by baierjan** [comment]
> If I understand this correctly this will always fail for every pull request that is created from a fork, right? Then we should not merge this.

Yes, you need to have `OPENQA_API_KEY` and `OPENQA_API_SECRET` for the pipeline to work. However, nothing prevents you to add your own credentials to your fork.

**Comment by Martchus** [comment]
> However, nothing prevents you to add your own credentials to your fork.

I don't think that would help here, though. At least when I initially tried it having credentials on my fork didn't work (except for PRs on my fork of course). I suppose secrets are just not transferred between the main repo and forks.

**Comment by perlpunk** [comment]
Just as an additioal explanation:
A workflow typically has two triggers: pull_request and push.

If you push to a branch to your fork, the push target of your fork will be triggered. If you have set secrets set in your fork, then this will work.
When you create a PR from your branch, then the pull_request target will be triggered. This runs inside the scope of the origin, but without the secrets, neither of the fork nor the origin. So the workflow will fail, even if your push workflow in your fork was successful.

**Comment by Martchus** [comment]
That makes unfortunately sense. I suppose we'll have to brainstorm a solution for that.

**Comment by baierjan** [comment]
Another approach which should support the fork-based development, the workflow will now run in the target context, ie. it should be able to use the credentials inside the main repository.

**Review by okurz — APPROVED** [review]
I guess we can only see after merge if this now works

**Review by Martchus — APPROVED** [review]

**Pull request merged**
